### PR TITLE
Atom Cheat Sheet

### PR DESCRIPTION
**Pull request by namandistro**
What does your Instant Answer do?

    Show all The Atom shortcut keys as cheat sheet.

What is the data source for your Instant Answer? (Provide a link if possible)

  https://bugsnag.com/blog/atom-editor-cheat-sheet

Why did you choose this data source?

    One of the Best and even recommended.

Are there any other alternative (better) data sources?

    Their are chances.

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

    Coder, Atom Users, Git users, Geeks etc.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

    Yes, https://duck.co/ideas/idea/6270/atom-cheatsheet

Are you having any problems? Do you need our help with anything?

    Yes, have a look at line 47, 61, 93, 97, 121.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

screenshot from 2015-08-20 13 52 33

-----
IA Page: https://duck.co/ia/view/github_atom_cheat_sheet

### CODE REVIEW

**Comment by moollaza** [comment]
Putting this one on hold for now -- I had already given permission to another (new) contributor to build this IA. When the other PR arrives I'll ask him to compare with this and add in any updates.

File: share/goodie/cheat_sheets/atom.json [hunk 1]
```diff
@@ -0,0 +1,172 @@
+{
+    "id":"atom_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: Please change to `github_atom_cheat_sheet`

File: share/goodie/cheat_sheets/atom.json [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            {
+                "key":"[⌘] [shift] [enter]",
+                "val":"Insert new line before current line"
+            },
+            {
+                "key":"[ctrl] [shift] [k]",
+                "val":"Delete current line"
+            },
+            {
+                "key":"[ctrl] [⌘] [(up/down)]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: let's use the ascii arrows for up/down/left/right . We provide example in the cheat sheet docs :+1:

Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: Also, please add padding around the alternatives: `[( up / down )]

File: share/goodie/cheat_sheets/atom.json [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                "val":"Delete text to beginning of word"
+            },
+            {
+                "key":"[option] [delete]",
+                "val":"Delete text to end of word"
+            }
+        ],
+        "Line Manipulation":[
+            {
+                "key":"[⌘] ] / [",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: You'll need to escape the square brackets here with `\\`: `"[⌘] {( \\] / \\[ )}"`

**Comment by namandistro** [comment]
@moollaza I made all the recommended changes but just have a look at ll the line mention where i need you help!

File: share/goodie/cheat_sheets/atom.json [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+                "val":"Set Bookmark"
+            },
+            {
+                "key":"[ctrl] [shift] [F2]",
+                "val":"Clear all bookmarks"
+            }
+        ],
+        "View/Window Manipulation":[
+            {
+                "key":"[⌘] [\\]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: if you're trying to show a single backslash, you need to use `\\\\` what you're really passing along to the Goodie parser is `\\` which we interpret as a literal `\`. The reason for the duplication is that JSON interprets `\\` as `\`.

File: share/goodie/cheat_sheets/atom.json [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+                "val":"Next result (find in files)"
+            },
+            {
+                "key":"[shift] [F4]",
+                "val":"Previous result (find in files)"
+            }
+        ],
+        "Folding":[
+            {
+                "key":"[ctrl] ([\\[]\/[\\]])",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: I think, for the sake of clarity/legibility, we should break this up into two key/val pairs so we can ditch the `( / )`

File: share/goodie/cheat_sheets/atom.json [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+                "val":"Delete text to beginning of word"
+            },
+            {
+                "key":"[option] [delete]",
+                "val":"Delete text to end of word"
+            }
+        ],
+        "Line Manipulation":[
+            {
+                "key":"[⌘] {( \\] / \\[ )}",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: I would break this into two k-v pairs as well

File: share/goodie/cheat_sheets/atom.json [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+            {
+                "key":"F2",
+                "val":"Goto next bookmark"
+            },
+            {
+                "key":"[ctrl] [k], [c]",
+                "val":"Goto current selection"
+            },
+            {
+                "key":"[ctrl] ([↑]/[↓])",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/atom.json]: These need spaces around them

**Comment by moollaza** [comment]
@namandistro I've made some final suggestions. We can merged after they're addressed,

File: share/goodie/cheat_sheets/atom.json (hunk 6 above)
Comment (namandistro) [inline share/goodie/cheat_sheets/atom.json]: Mention the line no here

File: share/goodie/cheat_sheets/atom.json (hunk 5 above)
Comment (namandistro) [inline share/goodie/cheat_sheets/atom.json]: i wanna add this  commands
 ctrl + ⌘ + [ / ]

option + shift + ⌘ + [ / ]

**Comment by moollaza** [comment]
@namandistro just saw that you committed to this PR. Please leave us a comment in the future so that we get a notification that you've made changes!

**Comment by moollaza** [comment]
LGTM :+1: 

For the future, please make sure to capitalize the key names: i.e. `[Ctrl] [Shift]`

**Pull request merged**